src/encodings/decoding.rs

```diff
@@ -470,14 +468,20 @@ impl<T: DataType> DeltaBitPackDecoder<T> {
   #[inline]
   fn load_deltas_in_mini_block(&mut self) -> Result<()> {
     self.deltas_in_mini_block.clear();
-
-    for _ in 0..self.values_current_mini_block {
-      // TODO: use SIMD to optimize this?
-      // TODO: we should be loading data eagerly like same decoder in parquet-mr
-      let delta = self.bit_reader
-        .get_value::<u64>(self.delta_bit_width as usize)
-        .ok_or(eof_err!("Not enough data to decode 'delta'"))?;
-      self.deltas_in_mini_block.push(delta);
+    if self.use_batch {
+      self.deltas_in_mini_block.resize(self.values_current_mini_block, T::T::default());
+      let loaded = self.bit_reader.get_batch::<T::T>(
+        &mut self.deltas_in_mini_block[..], self.delta_bit_width as usize
+      );
+      assert!(loaded == self.values_current_mini_block);
+    } else {
+      for _ in 0..self.values_current_mini_block {
+        // TODO: load one batch at a time similar to int32
```

Comment:
What currently stops us from doing this?

Reply:
`bit_packing::unpack32` only handles values up to 32 bit-width, but this is int64, so cannot be processed.